test/envtest/crds_envtest_test.go

```diff
@@ -325,6 +328,28 @@ func TestCRDValidations(t *testing.T) {
 				require.ErrorContains(t, err, "spec.healthchecks.passive.unhealthy.interval must not be set.")
 			},
 		},
+		{
+			name: "KongUpstreamPolicy - hashOn can only be set when algorithm is set to consistent-hashing",
```

Comment:
Can we add test cases for all the supported hash inputs + 1 for an unsupported to ensure the validation works?

Reply:
Added test cases for hash inputs as well as for all of the enums used in the CRD: https://github.com/Kong/kubernetes-ingress-controller/pull/4979/commits/d97c0e17f62a887af3b3ee3158f637c71e258981